en/flight_controller/beaglebone_blue.md

```diff
@@ -65,102 +58,88 @@ Currently running on a:
-```sh
-    make posix_bbblue_cross upload
-```
+> **Tip** This approach is recommended over [native build](#native_builds) due to speed of deployment and ease of use.
 
-To test the uploaded files on BeagleBone Blue board, run the following commands:
 
-```sh
-cd /home/debian/px4 
-sudo ./bin/px4 -s px4.config 
-```
-Currently librobotcontrol requires root access.
-
-Note that if you use a PX4 version before [Posix Shell](https://github.com/PX4/Firmware/pull/10173) 
-is available, run the following commands instead:
+1. First set up *rsync* (this is is used to transfer files from the development computer to the target board over a network - WiFi or Ethernet). 
+   - For *rsync* over SSH with key authentication, follow steps similar to those for [Raspberry Pi/Navio](../flight_controller/raspberry_pi_navio2.md)
+   - On the development computer, define the BeagleBone Blue board as `BBBluePX4` in **/etc/hosts**
+1. Run the following command to build and upload files:
+   ```sh
+   make posix_bbblue_cross upload
+   ```
 
+To test the uploaded files, run the following commands on the *BeagleBone Blue* board:
 ```sh
 cd /home/debian/px4 
-sudo px4 posix-configs/px4.config 
+sudo ./bin/px4 -s px4.config 
 ```
 
-### Native Builds (optional)
-
-You can run PX4 builds directly on the BeagleBone Blue if you desire. This is the *native*
-build. The other option is to run builds on a development computer which
-cross-compiles for the BeagleBone Blue, and pushes the PX4 executable binary directly to the
-BeagleBone Blue. This is the *cross-compiler* build, and the recommended one for developers
-due to speed of deployment and ease of use.
-
-For cross-compiling setups, you can skip this step.
+> **Note** Currently *librobotcontrol* requires root access.
 
-The steps below will setup the build system on the BeagleBone Blue to that required by PX4.
-Run these commands on the BeagleBone Blue itself!
+<span></span>
+> **Note** If you use a PX4 version before [Posix Shell](https://github.com/PX4/Firmware/pull/10173) is available, run the following commands instead:
+   ```sh
+   cd /home/debian/px4 
+   sudo px4 posix-configs/px4.config 
+  ```
 
-```sh
-sudo apt-get update
-sudo apt-get install cmake python-empy
-```
+## Native Builds (optional) {#native_builds}
 
-Then clone the Firmware directly onto the BeagleBone Blue.
+You can also natively build PX4 builds directly on the BeagleBone Blue.
 
-### Building the Code
+Run the following commands on the BeagleBone Blue (i.e. via SSH):
+1. Install dependencies:
+   ```sh
+   sudo apt-get update
+   sudo apt-get install cmake python-empy
+   ```
+1. Clone the PX4 Firmware directly onto the BeagleBone Blue.
+1. Continue with the [standard build system installation](https://dev.px4.io/en/setup/dev_env_linux.html).
 
-Continue with our [standard build system installation](https://dev.px4.io/en/setup/dev_env_linux.html).
 
-### Autostart during Boot
+## Autostart During Boot
```

Comment:
@UAV-Pilot Note, moved up to top level, because presumably this is something you want to do however you build.

Some concerns:
1. This is something done on the BBB right? Should say that. Also how do we modify these - via nano or similar over SSH?
2. There are two files described [/etc/rc.local] and *systemd*. Are these alternative different methods for autostart or two things you need to do to get systemd autostart to work?

Reply:
Sorry for the confusions.
1. Yes, on BBB. Yes, any editor like nano, vi over SSH.
2. They alternative methods, and only one method should be used. These 2 methods should not be applied at the same time.

Reply:
BTW, autostart is only needed when you actually fly a multicopter or a fixed wing airplane, etc. For most part of my porting work, I used manual start to test code change. Yes, for normal users who just use PX4, autostart is needed otherwise it's not convenient.